internal/impl/aws/input_kinesis_test.go

```diff
@@ -30,30 +37,56 @@ func (m *mockDynamoDbForKinesis) GetItemWithContext(aws.Context, *dynamodb.GetIt
 func (m *mockDynamoDbForKinesis) PutItem(*dynamodb.PutItemInput) (*dynamodb.PutItemOutput, error) {
 	return &dynamodb.PutItemOutput{}, nil
 }
+func (m *mockDynamoDbForKinesis) DeleteItemWithContext(aws.Context, *dynamodb.DeleteItemInput, ...request.Option) (*dynamodb.DeleteItemOutput, error) {
+	return &dynamodb.DeleteItemOutput{}, nil
+
+}
 
 type mockKinesisReader struct {
 	kinesisiface.KinesisAPI
 }
 
 func (m *mockKinesisReader) GetShardIteratorWithContext(aws.Context, *kinesis.GetShardIteratorInput, ...request.Option) (*kinesis.GetShardIteratorOutput, error) {
-	return &kinesis.GetShardIteratorOutput{}, nil
+	shardId := "2"
+	return &kinesis.GetShardIteratorOutput{ShardIterator: &shardId}, nil
+}
+
+func (m *mockKinesisReader) GetRecordsWithContext(aws.Context, *kinesis.GetRecordsInput, ...request.Option) (*kinesis.GetRecordsOutput, error) {
+	milisecondsBehind := int64(0)
+	return &kinesis.GetRecordsOutput{
+		ChildShards:        []*kinesis.ChildShard{},
+		MillisBehindLatest: &milisecondsBehind,
+		NextShardIterator:  nil,
+		Records:            []*kinesis.Record{},
+	}, nil
+}
+
+func NewAwsTestConfig(streams []string, streamsMaxShard []string) input.AWSKinesisConfig {
+	return input.AWSKinesisConfig{
+		Config:          session.NewConfig(),
+		Streams:         streams,
+		StreamsMaxShard: streamsMaxShard,
+		DynamoDB:        input.NewDynamoDBCheckpointConfig(),
+		CheckpointLimit: 1024,
+		CommitPeriod:    "5s",
+		LeasePeriod:     "30s",
+		RebalancePeriod: "30s",
+		StartFromOldest: true,
+		Batching:        batchconfig.NewConfig(),
```

Comment:
nit: Do we use these fields in our tests? Recommend to remove them if not required to avoid cluttering the test

Reply:
I have to pass in something to these data fields or it will complain 😓